packages/salesforcedx-vscode-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -53,13 +100,152 @@ export abstract class BaseNode extends TreeItem {
 
 export class CheckpointNode extends BaseNode {
   private readonly children: CheckpointInfoNode[] = [];
+  // These variables are necessary for the ApexExecutionOverlayAction
+  // and are not user editable. All user editable checkpoint pieces
+  // are stored on the children.
+  public executableEntityName: string;
+  public isDumpingHeap: boolean;
+  public line: number;
+  // Stored for convenience
+  public isTrigger: boolean;
+  // Source and URI are two things that the user cannot edit and
+  // it's debatable whether or not they need to be stored at all
+  public source: DebugProtocol.Source;
+  public uri: string;
+
+  constructor(
+    sourceInput: DebugProtocol.Source,
+    lineInput: number,
+    uriInput: string
+  ) {
+    super(
+      sourceInput.name + ':' + lineInput,
+      TreeItemCollapsibleState.Expanded
+    );
+
+    this.line = lineInput;
+    this.source = sourceInput;
+    this.uri = uriInput;
+    const sourceNameSplit = this.source.name!.split('.');
```

Comment:
I'm confused why you're using DebugProtocol.Source.Name for ExecutableEntityName. It's not a typeref which is why you have to prepend `__sfdc_trigger` for the trigger case. It also doesn't contain the namespace if the org is namespaced. The reason why we made ExecutableEntityName a typeref was because apexReplayDebug already has a mapping of URIs and typerefs, so you should use the info from that. That typeref is already in the correct format for ExecutableEntityName.

Reply:
Look at the source files on the disk for you project. These have to be in a specific format <ClassName>.cls for classes and <TriggerName>.trigger. These file names are the typeRef of the outermost type. source.name is the filename without the path, splitting off the extension leaves you with the typeRef name.

Reply:
The typeref in apexReplayDebug is from the apex language server which is our source of truth for typerefs. Are you saying you want this other thing here to figure typerefs? How are you going to handle namespace?

Reply:
Ignore my previous comment, I'm going to be using the typeref mapping in the breakpoint util.

Reply:
Yes, I made a mistake here. I'm going to be using the typerefMapping on the breakpointUtils to get the top level typeref.